src/pages/Home.tsx

```diff
@@ -21,10 +25,164 @@ import {jsx, SxStyleProp} from 'theme-ui';
+  const [posts, setPosts] = useState<IPost[]>([]);
+  const [fullyLoaded, setFullyLoaded] = useState<boolean>(false);
+  const [index, setIndex] = useState<number>(0);
+
+  
+
+  // TODO: get data from server
+
+  /**
+   * This uses to `fetch` API to request more posts from the server.  
+   * This is used to ensure that the server can dynamically load posts upon request and
+   * not load in everything at once.
+   * 
+   * For example, if the user wishes to load 30 more posts after seeing 50 posts,
+   * this will be `getPosts(50, 30)`. 
+   * This is zero-based index with lower bound (the index) inclusive. 
+   * 
+   * @param index The index to get posts from.
+   * @param amount The amount of posts to get.
+   * @return The newly retrieved posts.
+   */
+  const getMorePosts = (index: number, amount: number) => {
+    const newPosts: IPost[] = [
+      {
+        name: 'shari',
+        message: null,
+        photo: {
+          url: 'https://lh3.googleusercontent.com/d/13odY9Hp1vfGhUxV2yUfGbwAzZUJ2Gqhm=w400-h400?authuser=0',
+          name: 'lksdjf',
+        },
+      },
+      {
+        name: 'joeeeeeeeeeeeeee',
+        message: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt 
+ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris
+nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit
+esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident,
+sunt in culpa qui officia deserunt mollit anim id est laborum.
+`,
+        photo: {
+          url: 'https://lh3.googleusercontent.com/d/1dPt-MIi5eN_WxVkoQ2Y4QA4EE2I7bmKK=w400-h400?authuser=0',
+          name: 'ssssssssssss',
+        },
+      },
+      {
+        name: 'shari',
+        message: `hi sift if u didn't know already i actually love u so much. 
+im truly blessed to have met u back then, and im also rlly happy that we got closer. ur responsible and cool
+and u do a lot of cool things. but most of all, idk why, i just smile wheenver i look at u. i wanna see u happy,
+i wanna make u happy. i wanna stay forever with you. when i think about u, i just think of a smiling u, and its soooooo
+ooooooooo cute ^_^ i want u to find me cute, i want u to like me. u make me better and make my life better. u stick
+with me and we work things out well. our humours match and we're always on the same frequency. i wanna stick with u,
+ur such an amazing person and an amazing boyfriend (ur mine >:)) that is just too rare. i love u and i'll always do so ;)`,
+        photo: null,
+      },
+    ];
+    return newPosts;
+  };
+
+
+  /**
+   * Loads 30 more posts from the server.
+   */
+  const loadMorePosts = () => {
+    const amount = 30; // To test infinite scrolling, change this to like 2 (getMorePost returns 3 posts right now)
+    const newPosts = getMorePosts(index, amount);
+    if (newPosts.length < amount) {
```

Comment:
Could this ever be less than or equals to?

Reply:
yes, if the database didn't return back 30 posts since it ran out

Reply:
Ah hold on I finally got what that meant sorry my bad am monkee brain